tests/e2e/apiserver_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+    "context"
+    "fmt"
+    "os"
+    "path/filepath"
+    "time"
+
+    "bytes"
+    "io/ioutil"
+    yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+
+    . "github.com/onsi/ginkgo"
+    . "github.com/onsi/gomega"
+    batchv1 "k8s.io/api/batch/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    clientset "k8s.io/client-go/kubernetes"
+    restclientset "k8s.io/client-go/rest"
+    "k8s.io/client-go/tools/clientcmd"
+    "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+    restartDelaySec     = 30
+    testTimeoutSec      = 300
+)
+
+var _ = Describe("[apiserver] [Disruptive] the apiserver", func() {
+    f := framework.NewDefaultFramework("auth")
+    // TODO: figure out how to incorporate the framework!
```

Comment:
I'm honestly not sure about the framework--despite seeing it being used in other projects, I couldn't get it to work here. The kubeconfig is passed through the `--kubeconfig` flag and can be accessed using some package-level variables, but the ClientSet that *should* have been created in f.ClientSet just... didn't work for me when I tried it.

Besides, I don't think you can extract the configuration from the ClientSet anyways, so... yeah, this can probably be deleted.

Reply:
Ok, well if this works in prow then its fine for a first pass, we can look into the framework later/when needed.